src/protocols/secure_channel/CASESession.cpp

```diff
@@ -342,13 +373,22 @@ CHIP_ERROR CASESession::SendSigma1()
     ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(1), ByteSpan(mInitiatorRandom)));
     // Retrieve Session Identifier
     ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(2), GetLocalSessionId()));
-    // Generate a Destination Identifier
+    // Generate a Destination Identifier based on the node we are attempting to reach
     {
-        MutableByteSpan destinationIdSpan(destinationIdentifier);
         ReturnErrorCodeIf(mFabricInfo == nullptr, CHIP_ERROR_INCORRECT_STATE);
-        memcpy(mIPK, GetIPKList()->data(), sizeof(mIPK));
-        ReturnErrorOnFailure(
-            mFabricInfo->GenerateDestinationID(ByteSpan(mIPK), ByteSpan(mInitiatorRandom), GetPeerNodeId(), destinationIdSpan));
+
+        // Obtain originator IPK matching the fabric where we are trying to open a session. mIPK
+        // will be properly set thereafter.
+        ReturnErrorOnFailure(RecoverInitiatorIpk());
+
+        FabricId fabricId = mFabricInfo->GetFabricId();
+        uint8_t rootPubKeyBuf[Crypto::kP256_Point_Length];
+        Credentials::P256PublicKeySpan rootPubKeySpan(&rootPubKeyBuf[0]);
```

Comment:
```suggestion
        Credentials::P256PublicKeySpan rootPubKeySpan(rootPubKeyBuf);
```
which will helpfully ensure at compile time the buffer is big enough and whatnot...